### DISCUSSION TITLE
NOT A BUG - Question about dnscrypt-proxy.exe -list

### DISCUSSION THREAD

**Discussion by ghost** [DNSCrypt/dnscrypt-proxy#2166] [Show and tell]
Hi,

Running dnscrypt-proxy-win64-2.1.2 / Windows 7 x64 flawlessly as always.
My question concerns the _**-list**_ command-line parameter for dnscrypt-proxy.exe

I had created, I think it s for dnscrypt-proxy-win64-2.0.4.5, a batch file which would simply display the list of active resolvers:

```
@echo off
cmd.exe /c "dnscrypt-proxy.exe -list"
echo.
pause
```

It worked fine then, just tried it after years with dnscrypt-proxy-win64-2.1.2, issue, but I ignore on what version the issue appeared:
The issue is that the bat file still works but it breaks the dnscrypt-proxy.log simultaneously : only the first 2-3 line appear, the rest is simply removed.

This is not an issue strictly speaking . I've searched on this repository for "dnscrypt-proxy.exe -list" but haven't found any related comment. If I missed any info sorry for the bother.
Not important, but I'm puzzled. Thanks for any enlightenment!

**Comment by lifenjoiner** [comment]
It just prints to the console stdout rather than `dnscrypt-proxy.log`.
If you want put them in a file, use the redirection operator: `>`.

**Comment by ghost** [comment]
@lifenjoiner , thanks but not being a techie I don't understand your answer. I'm a non-techie using an elaborated dns encryption tool called DNScrypt-proxy and I manage to configure and run it flawlessly because its wiki pages are understandable by non-techies.

> If you want put them in a file, use the redirection operator: >.

How? Should the batch file state `cmd.exe /c "dnscrypt-proxy.exe -list" > list.txt` ?
I don't quite understand. I don't want to pipe the result to a file. I can get the list of the active resolvers just by displaying the dnscrypt-proxy.log, that's obvious. The fun with the above-mentioned dnscrypt-proxy.exe -list" batch file was that it'd just display those resolvers in a Command window. No problem to manage without that. I was only curious to know why this "dnscrypt-proxy.exe -list" when run from a batch file would now break the dnscrypt-proxy.log file as mentioned in my first comment.

I'll wait a few days then just close this comment of mine. Not sure it was ever pertinent to post it in the first place.

**Reply by lifenjoiner** [reply]
About the redirection, you're a good learner.

So, in another way, you are saying: the existing `dnscrypt-proxy.log` is overwritten by the `list` command.
You are running 2 `dnscrypt-proxy` processes, and they are writing to the same log file.
It seems to be an issue can be optimized ...

**Comment by ghost** [comment]
I may have poorly described what I'm facing which again is not an issue but only a puzzling problematic.
I am not concerned by the dnscrypt-proxy.log file, I mention it only because it is broken when I run the above mentioned batch file (named service-about_resolvers.bat and located of course in the DNScrypt-proxy folder):

```
@echo off
cmd.exe /c "dnscrypt-proxy.exe -list"
echo.
pause
```
1- My dnscrypt-proxy.log file this morning:
![1](https://user-images.githubusercontent.com/65161029/182552326-b2628b5f-0c45-4891-8681-cfb243876d47.png)

2- Running the bat file displays correctly the two active resolvers:
![2](https://user-images.githubusercontent.com/65161029/182552465-331307b7-8013-4c36-aa0c-78ffa7c883e9.png)

3- The dnscrypt-proxy.log file after:
![3](https://user-images.githubusercontent.com/65161029/182552573-5a66b0af-0f92-480d-9b34-5c64e4d616eb.png)

Obviously broken
So my wondering remains : Why does the batch file correctly display the active resolvers but at the same time break my dnscrypt-proxy.log file?
I repeat and emphasize on the fact that I have no idea with what version of dnscrypt-proxy this "issue" appeared. I had created the bat file when running dnscrypt-proxy 2.0.45 I think it was and just ran it again with dnscrypt-proxy 2.1.2, so the problem intervened between those two versions.

I'm willing to close this discussion because it's a lot of storming for a futile scheme.

**Comment by lifenjoiner** [comment]
Thanks for the details. It is caused by "the existing `dnscrypt-proxy.log` is overwritten by the `list` command".
PR https://github.com/DNSCrypt/dnscrypt-proxy/pull/2167 is submitted.
